pkg/cli/cmds/k3sopts.go

```diff
@@ -94,3 +83,61 @@ func commandFromK3S(cmd *cli.Command, flagOpts map[string]*K3SFlagOption) (*cli.
 	cmd.Flags = newFlags
 	return cmd, errs.Err()
 }
+
+// flagSetUsage receives a flag and a K3S flag option, parses
+// both and sets the necessary fields based on the underlying
+// flag type.
+func flagSetUsage(flag cli.Flag, opt *K3SFlagOption) {
+	v := reflect.ValueOf(flag).Elem()
+	if v.CanSet() {
+		switch t := flag.(type) {
+		case *cli.StringFlag:
+			t.Usage = opt.Usage
+		case *cli.StringSliceFlag:
+			t.Usage = opt.Usage
+		case *cli.BoolFlag:
+			t.Usage = opt.Usage
+		}
+	}
+}
+
+// flagSetDefault receives a flag and a K3S flag option, parses
+// both and sets the necessary fields based on the underlying
+// flag type.
+func flagSetDefault(flag cli.Flag, opt *K3SFlagOption) {
+	v := reflect.ValueOf(flag).Elem()
+	if v.CanSet() {
+		switch t := flag.(type) {
+		case *cli.StringFlag:
+			x := opt.Default
+			t.DefaultText = x
+			t.Destination = &x
+			t.Value = opt.Default
+		case *cli.StringSliceFlag:
+			t.DefaultText = opt.Default
+			t.Destination = &([]string{opt.Default})
+			t.Value = []string{opt.Default}
+		case *cli.BoolFlag:
+			t.DefaultText = opt.Default
+			t.Destination = &opt.Hide
```

Comment:
Why do bools store into the Hide flag? I thought Hide was for hiding it from help.

Reply:
I'm not really sure what you're asking. Can you clarify?

Reply:
As I read this, `t.Destination = &opt.Hide` means that the value of the flag will be stored in the Hide field of the option, which is used to determine if the option is hidden from the autogenerated `help` output. That seems wrong or at least counterintuitive.

Reply:
The address of the Hide field in the option struct is being stored in the Destination field on that type. I don't mind removing if not necessary but wanted to have it covered for thoroughness.

Reply:
Got it. This is some deep voodoo, thanks for walking me through it.